spec/modules/helpers.rb

```diff
@@ -71,4 +83,85 @@ def delete_category(category_name, category_description)
     expect(page).not_to have_selector("#category_#{category_id}")
   end
+end
+
+module DiscussionHelpers
+  include SharedMatchers
+  include DiscussionMatchers
+
+  def create_discussion(discussion_title, discussion_body, category_name = DEFAULT_CATEGORY_NAME)
+    create_category(category_name, CATEGORY_DESCRIPTION)
+
+    find_button "Create Discussion"
+    click_on "Create Discussion"
+    find_button "Submit"
+    fill_in("Title", with: discussion_title)
+    fill_in("Body", with: discussion_body)
+    click_on "Submit"
+    expect(page).to have_link_tree(category_name, DISCUSSION_TITLE)
+    expect(page).to have_correct_discussion_title_on_page(discussion_title)
+    expect(page).to have_correct_discussion_message_for_action("created")
+    expect(page).to have_expected_discussion_buttons
+    discussion = Discussion.find_by(title: DISCUSSION_TITLE, body: DISCUSSION_BODY)
+    within "#comment-section" do
+      expect(page).to have_write_comment_section
+    end
+    within "#user-info" do
+      expect(page).to have_correct_user_info(discussion)
+    end
+
+    # Check to make sure all the info for the newly created Discussion are correct
+    within "#link-tree" do
+      click_link category_name
+    end
+    expect(page).to have_the_expected_category_header("DISCUSSION REPLIES LAST POST")
+    within "#discussion-#{discussion.id}" do
+      find_by_id("discussion-#{discussion.id}-link")
+    end
+    within "#discussion-#{discussion.id}-info" do
+      expect(page).to have_text(discussion.user.login)
+      expect(page).to have_text(discussion.created_at.strftime("%m/%d/%Y %I:%M%p"))
+    end
+    within "#discussion-#{discussion.id}-replies .reply-counter" do
+      expect(page).to have_text(discussion.comments.count)
+    end
+
+    # Return back to the discussion#show for the newly created Discussion so the rest of the specs can work correctly
+    within "#discussion-#{discussion.id}-link" do
+      click_link DISCUSSION_TITLE
+    end
+  end
+
+  def edit_discussion(updated_title = "New Title", updated_body = "New Body", category_name = DEFAULT_CATEGORY_NAME, discussion_id)
+    create_discussion(DISCUSSION_TITLE, DISCUSSION_BODY)
+    discussion = Discussion.find_by(title: DISCUSSION_TITLE, body: DISCUSSION_BODY)
+
+    within "#discussion-title-buttons" do
+      click_on "Edit"
+    end
+    fill_in("Title", with: updated_title)
+    fill_in("Body", with: updated_body)
+    click_on "Submit"
+    expect(page).to have_link_tree(category_name, updated_title)
+    expect(page).to have_correct_discussion_title_on_page(updated_title)
+    expect(page).to have_correct_discussion_message_for_action("updated")
+    expect(page).to have_expected_discussion_buttons
+    within "#user-info" do
+      expect(page).to have_correct_user_info(discussion)
+    end
+    within "#comment-section" do
+      expect(page).to have_write_comment_section
+    end
+  end
+
+  def delete_category(discussion_title, discussion_body)
```

Comment:
This method signature does not match its function.  You're deleting a discussion, not a category.